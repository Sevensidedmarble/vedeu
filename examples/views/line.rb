#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    log '/tmp/vedeu_views_dsl.log'
    renderers(Vedeu::Renderers::Terminal.new,
              Vedeu::Renderers::File.new(filename: '/tmp/line.out'))
    run_once!
    standalone!
  end

  Vedeu.interface :test1_interface do
    border do
      title 'Test 1'
    end
    geometry do
      x  4
      y  3
      xn 34
      yn 8
    end
  end

  Vedeu.interface :help1_interface do
    geometry do
      x  use(:test1_interface).x
      y  use(:test1_interface).south
      xn use(:test1_interface).xn
      yn use(:test1_interface).south(3)
    end
  end

  Vedeu.interface :test2_interface do
    border do
      title 'Test 2'
    end
    geometry do
      x      use(:test1_interface).east(6)
      y      use(:test1_interface).y
      width  use(:test1_interface).width
      height use(:test1_interface).height
    end
  end

  Vedeu.interface :help2_interface do
    geometry do
      x  use(:test2_interface).x
      y  use(:test2_interface).south
      xn use(:test2_interface).xn
      yn use(:test2_interface).south(3)
    end
  end

  Vedeu.interface :test3_interface do
    border do
      title 'Test 3'
    end
    geometry do
      x  use(:help1_interface).x
      y  use(:help1_interface).south
      xn use(:help1_interface).xn
      yn use(:help1_interface).south(5)
    end
  end

  Vedeu.interface :help3_interface do
    geometry do
      x  use(:test3_interface).x
      y  use(:test3_interface).south
      xn use(:test3_interface).xn
      yn use(:test3_interface).south(5)
    end
  end

  Vedeu.interface :test4_interface do
    border do
      title 'Test 4'
    end
    geometry do
      x  use(:help2_interface).x
      y  use(:help2_interface).south
      xn use(:help2_interface).xn
      yn use(:help2_interface).south(5)
    end
  end

  Vedeu.interface :help4_interface do
    geometry do
      x  use(:test4_interface).x
      y  use(:test4_interface).south
      xn use(:test4_interface).xn
      yn use(:test4_interface).south(5)
    end
  end

  Vedeu.interface :test5_interface do
    border do
      title 'Test 5'
    end
    geometry do
      x  use(:help1_interface).x
      y  use(:help3_interface).south
      xn use(:help1_interface).xn
      yn use(:help3_interface).south(5)
    end
  end

  Vedeu.interface :help5_interface do
    geometry do
      x  use(:test5_interface).x
      y  use(:test5_interface).south
      xn use(:test5_interface).xn
      yn use(:test5_interface).south(5)
    end
  end

  Vedeu.render do
    view(:test1_interface) do
      background '#440000'
      line 'view->line', { foreground: '#ff0000' }
    end

    view(:help1_interface) do
      line 'A single line within a view.'
    end

    view(:test2_interface) do
      line 'view->line 1', { foreground: '#ff7f00' }
      line 'view->line 2', { foreground: '#ffff00' }
    end

    view(:help2_interface) do
      line 'Multiple lines within a view.'
      line '(No block)'
    end

    view(:test3_interface) do
      text 'view->text 1', { foreground: '#00ff00' }
      text 'view->text 2', { foreground: '#007fff' }
    end

    view(:help3_interface) do
      text 'Multiple `text` keywords within'
      text 'a view are treated as separate'
      text 'lines.'
    end

    view(:test4_interface) do
      line do
        text 'view->line 1->text', { foreground: '#5500cc' }
      end
      line do
        text 'view->line 2->text 1', { foreground: '#ff00ff' }
        text 'view->line 2->text 2', { foreground: '#ffffff' }
      end
    end

    view(:help4_interface) do
      line do
        foreground '#ff0000'
        text 'Line `block` syntax.'
      end
      line do
        background '#ffff00'
        foreground '#000000'
        text 'A single '
        text '`text` ', { background: '#000000', foreground: '#ffff00' }
        text 'is just a'
      end
      line do
        colour background: '#ff0000'
        text 'stream within a line, whilst'
      end
      line do
        colour foreground: '#006600'
        text 'multiple '
        text '`text`', { foreground: '#ff0000' }
        text 's are treated'
      end
      line do
        colour background: '#000066', foreground: '#ff9999'
        text 'as separate streams.'
      end
      line do
        foreground '#ff0000'
        text 'Line `block` syntax.', { background: '#00ff00' }
      end
    end

    view(:test5_interface) do
      lines do
        background '#000066'
        line 'view->lines->line 1', { foreground: '#00ff00' }
        line 'view->lines->line 2', { background: '#000000', foreground: '#007fff' }
        line 'view->lines->line 3', { foreground: '#55aa00' }
      end
    end

    view(:help5_interface) do
      lines do
        line 'Multiple `line` keywords within'
        line 'a `lines` block view are'
        line 'treated as separate lines.'
      end
    end
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

DSLApp.start
