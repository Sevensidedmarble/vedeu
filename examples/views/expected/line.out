[0m[0m[0m[3;4H[39m[49m(0l(B[3;5H(0q(B[3;6H [3;7HT[3;8He[3;9Hs[3;10Ht[3;11H [3;12H1[3;13H [3;14H(0q(B[3;15H(0q(B[3;16H(0q(B[3;17H(0q(B[3;18H(0q(B[3;19H(0q(B[3;20H(0q(B[3;21H(0q(B[3;22H(0q(B[3;23H(0q(B[3;24H(0q(B[3;25H(0q(B[3;26H(0q(B[3;27H(0q(B[3;28H(0q(B[3;29H(0q(B[3;30H(0q(B[3;31H(0q(B[3;32H(0q(B[3;33H(0q(B[3;34H(0k(B[3;40H(0l(B[3;41H(0q(B[3;42H [3;43HT[3;44He[3;45Hs[3;46Ht[3;47H [3;48H2[3;49H [3;50H(0q(B[3;51H(0q(B[3;52H(0q(B[3;53H(0q(B[3;54H(0q(B[3;55H(0q(B[3;56H(0q(B[3;57H(0q(B[3;58H(0q(B[3;59H(0q(B[3;60H(0q(B[3;61H(0q(B[3;62H(0q(B[3;63H(0q(B[3;64H(0q(B[3;65H(0q(B[3;66H(0q(B[3;67H(0q(B[3;68H(0q(B[3;69H(0q(B[3;70H(0k(B[0m[4;4H[39m[49m(0x(B[4;5H[38;5;196m[48;5;52mv[4;6Hi[4;7He[4;8Hw[4;9H-[4;10H>[4;11Hl[4;12Hi[4;13Hn[4;14He[4;15H[39m[49m [4;16H [4;17H [4;18H [4;19H [4;20H [4;21H [4;22H [4;23H [4;24H [4;25H [4;26H [4;27H [4;28H [4;29H [4;30H [4;31H [4;32H [4;33H [4;34H(0x(B[4;40H(0x(B[4;41H[38;5;208m[49mv[4;42Hi[4;43He[4;44Hw[4;45H-[4;46H>[4;47Hl[4;48Hi[4;49Hn[4;50He[4;51H [4;52H1[4;53H[39m[49m [4;54H [4;55H [4;56H [4;57H [4;58H [4;59H [4;60H [4;61H [4;62H [4;63H [4;64H [4;65H [4;66H [4;67H [4;68H [4;69H [4;70H(0x(B[0m[5;4H[39m[49m(0x(B[5;5H [5;6H [5;7H [5;8H [5;9H [5;10H [5;11H [5;12H [5;13H [5;14H [5;15H [5;16H [5;17H [5;18H [5;19H [5;20H [5;21H [5;22H [5;23H [5;24H [5;25H [5;26H [5;27H [5;28H [5;29H [5;30H [5;31H [5;32H [5;33H [5;34H(0x(B[5;40H(0x(B[5;41H[38;5;226m[49mv[5;42Hi[5;43He[5;44Hw[5;45H-[5;46H>[5;47Hl[5;48Hi[5;49Hn[5;50He[5;51H [5;52H2[5;53H[39m[49m [5;54H [5;55H [5;56H [5;57H [5;58H [5;59H [5;60H [5;61H [5;62H [5;63H [5;64H [5;65H [5;66H [5;67H [5;68H [5;69H [5;70H(0x(B[0m[6;4H[39m[49m(0x(B[6;5H [6;6H [6;7H [6;8H [6;9H [6;10H [6;11H [6;12H [6;13H [6;14H [6;15H [6;16H [6;17H [6;18H [6;19H [6;20H [6;21H [6;22H [6;23H [6;24H [6;25H [6;26H [6;27H [6;28H [6;29H [6;30H [6;31H [6;32H [6;33H [6;34H(0x(B[6;40H(0x(B[6;41H [6;42H [6;43H [6;44H [6;45H [6;46H [6;47H [6;48H [6;49H [6;50H [6;51H [6;52H [6;53H [6;54H [6;55H [6;56H [6;57H [6;58H [6;59H [6;60H [6;61H [6;62H [6;63H [6;64H [6;65H [6;66H [6;67H [6;68H [6;69H [6;70H(0x(B[0m[7;4H[39m[49m(0x(B[7;5H [7;6H [7;7H [7;8H [7;9H [7;10H [7;11H [7;12H [7;13H [7;14H [7;15H [7;16H [7;17H [7;18H [7;19H [7;20H [7;21H [7;22H [7;23H [7;24H [7;25H [7;26H [7;27H [7;28H [7;29H [7;30H [7;31H [7;32H [7;33H [7;34H(0x(B[7;40H(0x(B[7;41H [7;42H [7;43H [7;44H [7;45H [7;46H [7;47H [7;48H [7;49H [7;50H [7;51H [7;52H [7;53H [7;54H [7;55H [7;56H [7;57H [7;58H [7;59H [7;60H [7;61H [7;62H [7;63H [7;64H [7;65H [7;66H [7;67H [7;68H [7;69H [7;70H(0x(B[0m[8;4H[39m[49m(0m(B[8;5H(0q(B[8;6H(0q(B[8;7H(0q(B[8;8H(0q(B[8;9H(0q(B[8;10H(0q(B[8;11H(0q(B[8;12H(0q(B[8;13H(0q(B[8;14H(0q(B[8;15H(0q(B[8;16H(0q(B[8;17H(0q(B[8;18H(0q(B[8;19H(0q(B[8;20H(0q(B[8;21H(0q(B[8;22H(0q(B[8;23H(0q(B[8;24H(0q(B[8;25H(0q(B[8;26H(0q(B[8;27H(0q(B[8;28H(0q(B[8;29H(0q(B[8;30H(0q(B[8;31H(0q(B[8;32H(0q(B[8;33H(0q(B[8;34H(0j(B[8;40H(0m(B[8;41H(0q(B[8;42H(0q(B[8;43H(0q(B[8;44H(0q(B[8;45H(0q(B[8;46H(0q(B[8;47H(0q(B[8;48H(0q(B[8;49H(0q(B[8;50H(0q(B[8;51H(0q(B[8;52H(0q(B[8;53H(0q(B[8;54H(0q(B[8;55H(0q(B[8;56H(0q(B[8;57H(0q(B[8;58H(0q(B[8;59H(0q(B[8;60H(0q(B[8;61H(0q(B[8;62H(0q(B[8;63H(0q(B[8;64H(0q(B[8;65H(0q(B[8;66H(0q(B[8;67H(0q(B[8;68H(0q(B[8;69H(0q(B[8;70H(0j(B[0m[9;4H[39m[49mA[9;5H [9;6Hs[9;7Hi[9;8Hn[9;9Hg[9;10Hl[9;11He[9;12H [9;13Hl[9;14Hi[9;15Hn[9;16He[9;17H [9;18Hw[9;19Hi[9;20Ht[9;21Hh[9;22Hi[9;23Hn[9;24H [9;25Ha[9;26H [9;27Hv[9;28Hi[9;29He[9;30Hw[9;31H.[9;32H [9;33H [9;34H [9;40HM[9;41Hu[9;42Hl[9;43Ht[9;44Hi[9;45Hp[9;46Hl[9;47He[9;48H [9;49Hl[9;50Hi[9;51Hn[9;52He[9;53Hs[9;54H [9;55Hw[9;56Hi[9;57Ht[9;58Hh[9;59Hi[9;60Hn[9;61H [9;62Ha[9;63H [9;64Hv[9;65Hi[9;66He[9;67Hw[9;68H.[9;69H [9;70H [0m[10;4H[39m[49m [10;5H [10;6H [10;7H [10;8H [10;9H [10;10H [10;11H [10;12H [10;13H [10;14H [10;15H [10;16H [10;17H [10;18H [10;19H [10;20H [10;21H [10;22H [10;23H [10;24H [10;25H [10;26H [10;27H [10;28H [10;29H [10;30H [10;31H [10;32H [10;33H [10;34H [10;40H([10;41HN[10;42Ho[10;43H [10;44Hb[10;45Hl[10;46Ho[10;47Hc[10;48Hk[10;49H)[10;50H [10;51H [10;52H [10;53H [10;54H [10;55H [10;56H [10;57H [10;58H [10;59H [10;60H [10;61H [10;62H [10;63H [10;64H [10;65H [10;66H [10;67H [10;68H [10;69H [10;70H [0m[11;4H[39m[49m [11;5H [11;6H [11;7H [11;8H [11;9H [11;10H [11;11H [11;12H [11;13H [11;14H [11;15H [11;16H [11;17H [11;18H [11;19H [11;20H [11;21H [11;22H [11;23H [11;24H [11;25H [11;26H [11;27H [11;28H [11;29H [11;30H [11;31H [11;32H [11;33H [11;34H [11;40H [11;41H [11;42H [11;43H [11;44H [11;45H [11;46H [11;47H [11;48H [11;49H [11;50H [11;51H [11;52H [11;53H [11;54H [11;55H [11;56H [11;57H [11;58H [11;59H [11;60H [11;61H [11;62H [11;63H [11;64H [11;65H [11;66H [11;67H [11;68H [11;69H [11;70H [0m[12;4H[39m[49m(0l(B[12;5H(0q(B[12;6H [12;7HT[12;8He[12;9Hs[12;10Ht[12;11H [12;12H3[12;13H [12;14H(0q(B[12;15H(0q(B[12;16H(0q(B[12;17H(0q(B[12;18H(0q(B[12;19H(0q(B[12;20H(0q(B[12;21H(0q(B[12;22H(0q(B[12;23H(0q(B[12;24H(0q(B[12;25H(0q(B[12;26H(0q(B[12;27H(0q(B[12;28H(0q(B[12;29H(0q(B[12;30H(0q(B[12;31H(0q(B[12;32H(0q(B[12;33H(0q(B[12;34H(0k(B[12;40H(0l(B[12;41H(0q(B[12;42H [12;43HT[12;44He[12;45Hs[12;46Ht[12;47H [12;48H4[12;49H [12;50H(0q(B[12;51H(0q(B[12;52H(0q(B[12;53H(0q(B[12;54H(0q(B[12;55H(0q(B[12;56H(0q(B[12;57H(0q(B[12;58H(0q(B[12;59H(0q(B[12;60H(0q(B[12;61H(0q(B[12;62H(0q(B[12;63H(0q(B[12;64H(0q(B[12;65H(0q(B[12;66H(0q(B[12;67H(0q(B[12;68H(0q(B[12;69H(0q(B[12;70H(0k(B[0m[13;4H[39m[49m(0x(B[13;5H[38;5;46m[49mv[13;6Hi[13;7He[13;8Hw[13;9H-[13;10H>[13;11Ht[13;12He[13;13Hx[13;14Ht[13;15H [13;16H1[13;17H[39m[49m [13;18H [13;19H [13;20H [13;21H [13;22H [13;23H [13;24H [13;25H [13;26H [13;27H [13;28H [13;29H [13;30H [13;31H [13;32H [13;33H [13;34H(0x(B[13;40H(0x(B[13;41H[38;5;56m[49mv[13;42Hi[13;43He[13;44Hw[13;45H-[13;46H>[13;47Hl[13;48Hi[13;49Hn[13;50He[13;51H [13;52H1[13;53H-[13;54H>[13;55Ht[13;56He[13;57Hx[13;58Ht[13;59H[39m[49m [13;60H [13;61H [13;62H [13;63H [13;64H [13;65H [13;66H [13;67H [13;68H [13;69H [13;70H(0x(B[0m[14;4H[39m[49m(0x(B[14;5H[38;5;33m[49mv[14;6Hi[14;7He[14;8Hw[14;9H-[14;10H>[14;11Ht[14;12He[14;13Hx[14;14Ht[14;15H [14;16H2[14;17H[39m[49m [14;18H [14;19H [14;20H [14;21H [14;22H [14;23H [14;24H [14;25H [14;26H [14;27H [14;28H [14;29H [14;30H [14;31H [14;32H [14;33H [14;34H(0x(B[14;40H(0x(B[14;41H[38;5;201m[49mv[14;42Hi[14;43He[14;44Hw[14;45H-[14;46H>[14;47Hl[14;48Hi[14;49Hn[14;50He[14;51H [14;52H2[14;53H-[14;54H>[14;55Ht[14;56He[14;57Hx[14;58Ht[14;59H [14;60H1[14;61H[38;5;231m[49mv[14;62Hi[14;63He[14;64Hw[14;65H-[14;66H>[14;67Hl[14;68Hi[14;69Hn[14;70H[39m[49m(0x(B[0m[15;4H[39m[49m(0x(B[15;5H [15;6H [15;7H [15;8H [15;9H [15;10H [15;11H [15;12H [15;13H [15;14H [15;15H [15;16H [15;17H [15;18H [15;19H [15;20H [15;21H [15;22H [15;23H [15;24H [15;25H [15;26H [15;27H [15;28H [15;29H [15;30H [15;31H [15;32H [15;33H [15;34H(0x(B[15;40H(0x(B[15;41H [15;42H [15;43H [15;44H [15;45H [15;46H [15;47H [15;48H [15;49H [15;50H [15;51H [15;52H [15;53H [15;54H [15;55H [15;56H [15;57H [15;58H [15;59H [15;60H [15;61H [15;62H [15;63H [15;64H [15;65H [15;66H [15;67H [15;68H [15;69H [15;70H(0x(B[0m[16;4H[39m[49m(0m(B[16;5H(0q(B[16;6H(0q(B[16;7H(0q(B[16;8H(0q(B[16;9H(0q(B[16;10H(0q(B[16;11H(0q(B[16;12H(0q(B[16;13H(0q(B[16;14H(0q(B[16;15H(0q(B[16;16H(0q(B[16;17H(0q(B[16;18H(0q(B[16;19H(0q(B[16;20H(0q(B[16;21H(0q(B[16;22H(0q(B[16;23H(0q(B[16;24H(0q(B[16;25H(0q(B[16;26H(0q(B[16;27H(0q(B[16;28H(0q(B[16;29H(0q(B[16;30H(0q(B[16;31H(0q(B[16;32H(0q(B[16;33H(0q(B[16;34H(0j(B[16;40H(0m(B[16;41H(0q(B[16;42H(0q(B[16;43H(0q(B[16;44H(0q(B[16;45H(0q(B[16;46H(0q(B[16;47H(0q(B[16;48H(0q(B[16;49H(0q(B[16;50H(0q(B[16;51H(0q(B[16;52H(0q(B[16;53H(0q(B[16;54H(0q(B[16;55H(0q(B[16;56H(0q(B[16;57H(0q(B[16;58H(0q(B[16;59H(0q(B[16;60H(0q(B[16;61H(0q(B[16;62H(0q(B[16;63H(0q(B[16;64H(0q(B[16;65H(0q(B[16;66H(0q(B[16;67H(0q(B[16;68H(0q(B[16;69H(0q(B[16;70H(0j(B[0m[17;4H[39m[49mM[17;5Hu[17;6Hl[17;7Ht[17;8Hi[17;9Hp[17;10Hl[17;11He[17;12H [17;13H`[17;14Ht[17;15He[17;16Hx[17;17Ht[17;18H`[17;19H [17;20Hk[17;21He[17;22Hy[17;23Hw[17;24Ho[17;25Hr[17;26Hd[17;27Hs[17;28H [17;29Hw[17;30Hi[17;31Ht[17;32Hh[17;33Hi[17;34Hn[17;40H[38;5;196m[49mL[17;41Hi[17;42Hn[17;43He[17;44H [17;45H`[17;46Hb[17;47Hl[17;48Ho[17;49Hc[17;50Hk[17;51H`[17;52H [17;53Hs[17;54Hy[17;55Hn[17;56Ht[17;57Ha[17;58Hx[17;59H.[17;60H[39m[49m [17;61H [17;62H [17;63H [17;64H [17;65H [17;66H [17;67H [17;68H [17;69H [17;70H [0m[18;4H[39m[49ma[18;5H [18;6Hv[18;7Hi[18;8He[18;9Hw[18;10H [18;11Ha[18;12Hr[18;13He[18;14H [18;15Ht[18;16Hr[18;17He[18;18Ha[18;19Ht[18;20He[18;21Hd[18;22H [18;23Ha[18;24Hs[18;25H [18;26Hs[18;27He[18;28Hp[18;29Ha[18;30Hr[18;31Ha[18;32Ht[18;33He[18;34H [18;40H[38;5;16m[48;5;226mA[18;41H [18;42Hs[18;43Hi[18;44Hn[18;45Hg[18;46Hl[18;47He[18;48H [18;49H[38;5;226m[48;5;16m`[18;50Ht[18;51He[18;52Hx[18;53Ht[18;54H`[18;55H [18;56H[38;5;16m[48;5;226mi[18;57Hs[18;58H [18;59Hj[18;60Hu[18;61Hs[18;62Ht[18;63H [18;64Ha[18;65H[39m[49m [18;66H [18;67H [18;68H [18;69H [18;70H [0m[19;4H[39m[49ml[19;5Hi[19;6Hn[19;7He[19;8Hs[19;9H.[19;10H [19;11H [19;12H [19;13H [19;14H [19;15H [19;16H [19;17H [19;18H [19;19H [19;20H [19;21H [19;22H [19;23H [19;24H [19;25H [19;26H [19;27H [19;28H [19;29H [19;30H [19;31H [19;32H [19;33H [19;34H [19;40H[39m[48;5;196ms[19;41Ht[19;42Hr[19;43He[19;44Ha[19;45Hm[19;46H [19;47Hw[19;48Hi[19;49Ht[19;50Hh[19;51Hi[19;52Hn[19;53H [19;54Ha[19;55H [19;56Hl[19;57Hi[19;58Hn[19;59He[19;60H,[19;61H [19;62Hw[19;63Hh[19;64Hi[19;65Hl[19;66Hs[19;67Ht[19;68H[39m[49m [19;69H [19;70H [0m[20;4H[39m[49m [20;5H [20;6H [20;7H [20;8H [20;9H [20;10H [20;11H [20;12H [20;13H [20;14H [20;15H [20;16H [20;17H [20;18H [20;19H [20;20H [20;21H [20;22H [20;23H [20;24H [20;25H [20;26H [20;27H [20;28H [20;29H [20;30H [20;31H [20;32H [20;33H [20;34H [20;40H[38;5;28m[49mm[20;41Hu[20;42Hl[20;43Ht[20;44Hi[20;45Hp[20;46Hl[20;47He[20;48H [20;49H[38;5;196m[49m`[20;50Ht[20;51He[20;52Hx[20;53Ht[20;54H`[20;55H[38;5;28m[49ms[20;56H [20;57Ha[20;58Hr[20;59He[20;60H [20;61Ht[20;62Hr[20;63He[20;64Ha[20;65Ht[20;66He[20;67Hd[20;68H[39m[49m [20;69H [20;70H [0m[21;4H[39m[49m [21;5H [21;6H [21;7H [21;8H [21;9H [21;10H [21;11H [21;12H [21;13H [21;14H [21;15H [21;16H [21;17H [21;18H [21;19H [21;20H [21;21H [21;22H [21;23H [21;24H [21;25H [21;26H [21;27H [21;28H [21;29H [21;30H [21;31H [21;32H [21;33H [21;34H [21;40H[38;5;217m[48;5;18ma[21;41Hs[21;42H [21;43Hs[21;44He[21;45Hp[21;46Ha[21;47Hr[21;48Ha[21;49Ht[21;50He[21;51H [21;52Hs[21;53Ht[21;54Hr[21;55He[21;56Ha[21;57Hm[21;58Hs[21;59H.[21;60H[39m[49m [21;61H [21;62H [21;63H [21;64H [21;65H [21;66H [21;67H [21;68H [21;69H [21;70H [0m[22;4H[39m[49m(0l(B[22;5H(0q(B[22;6H [22;7HT[22;8He[22;9Hs[22;10Ht[22;11H [22;12H5[22;13H [22;14H(0q(B[22;15H(0q(B[22;16H(0q(B[22;17H(0q(B[22;18H(0q(B[22;19H(0q(B[22;20H(0q(B[22;21H(0q(B[22;22H(0q(B[22;23H(0q(B[22;24H(0q(B[22;25H(0q(B[22;26H(0q(B[22;27H(0q(B[22;28H(0q(B[22;29H(0q(B[22;30H(0q(B[22;31H(0q(B[22;32H(0q(B[22;33H(0q(B[22;34H(0k(B[0m[23;4H[39m[49m(0x(B[23;5H[38;5;46m[48;5;18mv[23;6Hi[23;7He[23;8Hw[23;9H-[23;10H>[23;11Hl[23;12Hi[23;13Hn[23;14He[23;15Hs[23;16H-[23;17H>[23;18Hl[23;19Hi[23;20Hn[23;21He[23;22H [23;23H1[23;24H[39m[49m [23;25H [23;26H [23;27H [23;28H [23;29H [23;30H [23;31H [23;32H [23;33H [23;34H(0x(B[0m[24;4H[39m[49m(0x(B[24;5H[38;5;33m[48;5;16mv[24;6Hi[24;7He[24;8Hw[24;9H-[24;10H>[24;11Hl[24;12Hi[24;13Hn[24;14He[24;15Hs[24;16H-[24;17H>[24;18Hl[24;19Hi[24;20Hn[24;21He[24;22H [24;23H2[24;24H[39m[49m [24;25H [24;26H [24;27H [24;28H [24;29H [24;30H [24;31H [24;32H [24;33H [24;34H(0x(B[0m[25;4H[39m[49m(0x(B[25;5H[38;5;70m[48;5;18mv[25;6Hi[25;7He[25;8Hw[25;9H-[25;10H>[25;11Hl[25;12Hi[25;13Hn[25;14He[25;15Hs[25;16H-[25;17H>[25;18Hl[25;19Hi[25;20Hn[25;21He[25;22H [25;23H3[25;24H[39m[49m [25;25H [25;26H [25;27H [25;28H [25;29H [25;30H [25;31H [25;32H [25;33H [25;34H(0x(B[0m[26;4H[39m[49m(0m(B[26;5H(0q(B[26;6H(0q(B[26;7H(0q(B[26;8H(0q(B[26;9H(0q(B[26;10H(0q(B[26;11H(0q(B[26;12H(0q(B[26;13H(0q(B[26;14H(0q(B[26;15H(0q(B[26;16H(0q(B[26;17H(0q(B[26;18H(0q(B[26;19H(0q(B[26;20H(0q(B[26;21H(0q(B[26;22H(0q(B[26;23H(0q(B[26;24H(0q(B[26;25H(0q(B[26;26H(0q(B[26;27H(0q(B[26;28H(0q(B[26;29H(0q(B[26;30H(0q(B[26;31H(0q(B[26;32H(0q(B[26;33H(0q(B[26;34H(0j(B[0m[27;4H[39m[49mM[27;5Hu[27;6Hl[27;7Ht[27;8Hi[27;9Hp[27;10Hl[27;11He[27;12H [27;13H`[27;14Hl[27;15Hi[27;16Hn[27;17He[27;18H`[27;19H [27;20Hk[27;21He[27;22Hy[27;23Hw[27;24Ho[27;25Hr[27;26Hd[27;27Hs[27;28H [27;29Hw[27;30Hi[27;31Ht[27;32Hh[27;33Hi[27;34Hn[0m[28;4H[39m[49ma[28;5H [28;6H`[28;7Hl[28;8Hi[28;9Hn[28;10He[28;11Hs[28;12H`[28;13H [28;14Hb[28;15Hl[28;16Ho[28;17Hc[28;18Hk[28;19H [28;20Hv[28;21Hi[28;22He[28;23Hw[28;24H [28;25Ha[28;26Hr[28;27He[28;28H [28;29H [28;30H [28;31H [28;32H [28;33H [28;34H [0m[29;4H[39m[49mt[29;5Hr[29;6He[29;7Ha[29;8Ht[29;9He[29;10Hd[29;11H [29;12Ha[29;13Hs[29;14H [29;15Hs[29;16He[29;17Hp[29;18Ha[29;19Hr[29;20Ha[29;21Ht[29;22He[29;23H [29;24Hl[29;25Hi[29;26Hn[29;27He[29;28Hs[29;29H.[29;30H [29;31H [29;32H [29;33H [29;34H [0m[30;4H[39m[49m [30;5H [30;6H [30;7H [30;8H [30;9H [30;10H [30;11H [30;12H [30;13H [30;14H [30;15H [30;16H [30;17H [30;18H [30;19H [30;20H [30;21H [30;22H [30;23H [30;24H [30;25H [30;26H [30;27H [30;28H [30;29H [30;30H [30;31H [30;32H [30;33H [30;34H [0m[31;4H[39m[49m [31;5H [31;6H [31;7H [31;8H [31;9H [31;10H [31;11H [31;12H [31;13H [31;14H [31;15H [31;16H [31;17H [31;18H [31;19H [31;20H [31;21H [31;22H [31;23H [31;24H [31;25H [31;26H [31;27H [31;28H [31;29H [31;30H [31;31H [31;32H [31;33H [31;34H [0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m
